### DISCUSSION TITLE
Unsure how to upgrade to latest version. What is SessionStorage these days?

### DISCUSSION THREAD

**Discussion by Pinwheeler** [sergiodxa/remix-auth#315] [Q&A] [answered]
I'm liking a lot of simplicity that flows out of the latest version of `remix-auth: 4.1.0` and `remix-auth-form: 3.0.0` however, I can't quite figure out the sticks.

I'm upgrading from `remix-auth: 3.7.0` and `remix-auth-form: 1.5.0` where we are creating our own session storage object. Mine looks like this

```
import { createCookieSessionStorage } from "@remix-run/node";

const secret = process.env.SESSION_SECRET || "default_secret";

// export the whole sessionStorage object
export const sessionStorage = createCookieSessionStorage({
  cookie: {
    name: "_session", // use any name you want here
    sameSite: "lax", // this helps with CSRF
    path: "/", // remember to add this so the cookie will work in all routes
    httpOnly: true, // for security reasons, make this cookie http only
    secrets: [secret], // replace this with an actual secret
    secure: process.env.NODE_ENV === "production", // enable this in prod only
  },
});

// you can also export the methods individually for your own usage
export const { getSession, commitSession, destroySession } = sessionStorage;

```

It seems like in the latest version of auth that we don't need to do that, instead favoring https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage

Overall, I'm just curious what's going on. Specifically, I'm having an issue where upgrading is causing some infinite loop spinning. I'm sure I can figure that out if I can understand what is going on with the changes I'm mentioning.

A great place to start might be explaining this quote:

> The sessionStorage can be created using React Router's session storage hepler, is up to you to decide what session storage mechanism you want to use, or how you plan to keep the user data after authentication, maybe you just need a plain cookie.

**Comment by sergiodxa** [comment, marked as answer]
I'm not sure why you think it recommend using browser's sessionStorage object, the quote mentions to use React Router session storage objects, the same you used before.

By you deciding what to use is because there are more than just createCookieSessionStorage, eg you could use createWorkerKVSessionStorage when using CF, or a custom one with createSessionStorage.

**Reply by Pinwheeler** [reply]
okay- I guess it's not clear to me that `@remix-run/node > createCookieSessionStorage` is equivalent to "React Router session storage helper"

1. I struggled to import some kind of "session storage helper" and instead opted to 
2. just use the code exactly as written which
3. lacking explicit imports
4. defaults to assuming we're using the browser's sessionStorage object
